penty/pentypes/builtins.py

```diff
@@ -472,21 +479,246 @@ def set_init(elts_ty=None):
     next_ty = Types[iter_ty]['__next__'](iter_ty)
     return _Set[next_ty]
 
+def set_ior(self_ty, other_ty):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not issubclass(other_ty, set):
+        raise TypeError
+    self_ty.__args__[0] |= other_ty.__args__[0]
+    return self_ty
+
+def set_isub(self_ty, other_ty):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not issubclass(other_ty, set):
+        raise TypeError
+    return self_ty
+
+def set_ixor(self_ty, other_ty):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not issubclass(other_ty, set):
+        raise TypeError
+    self_ty.__args__[0] |= other_ty.__args__[0]
+    return self_ty
+
+def set_len(self_ty):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    if not self_ty.__args__[0]:
+        return _Cst[0]
+    return int
+
+def set_iter(self_ty):
+    if not issubclass(self_ty, set):
+        raise TypeError
+    return {_SetIterator[ty] for ty in self_ty.__args__[0]}
```

Comment:
Shouldn't is be `_SetIterator[self_ty.__args__[0]]` ?

If the set can contain values of multiple types, an iterator will yield values of all of those types, not values of one of the types

Reply:
Yes!